### PR TITLE
membership: api group membership endpoint

### PR DESCRIPTION
**Pull request by DTabidze**
affects #98

### CODE REVIEW

File: core/auth/memberships/main.go [hunk 1]
```diff
@@ -311,6 +314,62 @@ func (s *SQLiteStore) GetAvailableGroupsAsChild(group string) ([]string, error)
...
+	}
+	for _, parentGroup := range parentGroups {
+		if err := s.getParentGroups(parentGroup, allGroups); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *SQLiteStore) getParentGroupsFromDB(group string) ([]string, error) {
```
Comment (giolekva) [inline core/auth/memberships/main.go:352]: Rename to `GetGroupsGroupBelongsTo`

Also rename `GetMembershipGroups` to `GetGroupsUserBelongsTo`

File: core/auth/memberships/main.go [hunk 2]
```diff
@@ -311,6 +314,62 @@ func (s *SQLiteStore) GetAvailableGroupsAsChild(group string) ([]string, error)
 	return availableGroups, nil
 }
 
+func (s *SQLiteStore) GetAllAssociatedGroups(user string) ([]string, error) {
```
Comment (giolekva) [inline core/auth/memberships/main.go:317]: Rename to `GetAllTransitiveGroupsForUser`

**Review by giolekva — CHANGES_REQUESTED** [review]

File: core/auth/memberships/main.go (hunk 2 above)
Comment (DTabidze) [reply core/auth/memberships/main.go]: Done.

File: core/auth/memberships/main.go (hunk 1 above)
Comment (DTabidze) [reply core/auth/memberships/main.go]: Done.

File: core/auth/memberships/main.go [hunk 3]
```diff
@@ -526,6 +590,28 @@ func (s *Server) addChildGroupHandler(w http.ResponseWriter, r *http.Request) {
...
+	if !ok {
+		http.Error(w, "Username parameter is required", http.StatusBadRequest)
+		return
+	}
+	associatedGroups, err := s.store.GetAllTransitiveGroupsForUser(user)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	response, err := json.Marshal(map[string]interface{}{"memberOf": associatedGroups})
```
Comment (giolekva) [inline core/auth/memberships/main.go:606]:
```go
type UserInfo struct {
  MemberOf []string `json:"memberOf"`
}

json.NewEncoder(w).Encode(UserInfo{associatedGroups})
```

File: core/auth/memberships/main.go [hunk 4]
```diff
@@ -526,6 +590,28 @@ func (s *Server) addChildGroupHandler(w http.ResponseWriter, r *http.Request) {
...
 
+func (s *Server) apiMemberOfHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Println("apiMemberOfHandler called")
+	vars := mux.Vars(r)
+	user, ok := vars["username"]
+	if !ok {
+		http.Error(w, "Username parameter is required", http.StatusBadRequest)
+		return
+	}
+	associatedGroups, err := s.store.GetAllTransitiveGroupsForUser(user)
```
Comment (giolekva) [inline core/auth/memberships/main.go:601]: rename to `transitiveGroups`

File: core/auth/memberships/main.go [hunk 5]
```diff
@@ -526,6 +590,28 @@ func (s *Server) addChildGroupHandler(w http.ResponseWriter, r *http.Request) {
 	http.Redirect(w, r, "/group/"+parentGroup, http.StatusSeeOther)
 }
 
+func (s *Server) apiMemberOfHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Println("apiMemberOfHandler called")
```
Comment (giolekva) [inline core/auth/memberships/main.go:594]: rm debug line

File: core/auth/memberships/main.go [hunk 6]
```diff
@@ -526,6 +590,28 @@ func (s *Server) addChildGroupHandler(w http.ResponseWriter, r *http.Request) {
...
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	response, err := json.Marshal(map[string]interface{}{"memberOf": associatedGroups})
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
```
Comment (giolekva) [inline core/auth/memberships/main.go:611]: Move this up before json encoding

**Review by giolekva — CHANGES_REQUESTED** [review]

File: core/auth/memberships/main.go (hunk 3 above)
Comment (DTabidze) [reply core/auth/memberships/main.go]: Done.

File: core/auth/memberships/main.go (hunk 6 above)
Comment (DTabidze) [reply core/auth/memberships/main.go]: Done.

**Review by giolekva — APPROVED** [review]

**Pull request merged**